test.js

```diff
@@ -1,201 +1,114 @@
 "use strict";
 
-var test = require("tape");
-var check = require("./");
-
-function getMockedGetVersionsOptions(versions) {
-  return {
-    getVersion: function(name, callback) {
-      process.nextTick(function() {
-        if (versions[name] instanceof Error) {
-          callback(null, {
-            error: versions[name],
-          });
-        }
-        else {
-          callback(null, {
-            version: versions[name],
-          });
-        }
-      });
-    },
-  };
-}
-
-test("simple call", function(t) {
-  check(function(err, result) {
-    t.ifError(err);
-    t.ok(result.versions.node);
-    t.ok(result.versions.node.version);
-    t.ok(result.versions.npm);
-    t.ok(result.versions.npm.version);
-    t.ok(result.versions.yarn);
-    t.ok(result.isSatisfied);
-    t.end();
-  });
-});
+const test = require("ava").cb;
 
-test("positive node result", function(t) {
-  var version = "7.0.0";
-  var wanted = {
-    node: version,
-  };
-  var options = getMockedGetVersionsOptions({
-    node: version,
-  });
-  check(wanted, options, function(err, result) {
-    t.ifError(err);
-    t.ok(result.versions.node);
-    t.ok(result.versions.node.version);
-    t.equal(result.versions.node.version.raw, version);
-    t.equal(result.versions.node.wanted.range, version);
-    t.equal(result.versions.node.isSatisfied, true);
-    t.equal(result.isSatisfied, true);
-    t.end();
-  });
-});
+const check = require(".");
 
-test("negative node result", function(t) {
-  var version = "7.0.0";
-  var wanted = {
-    node: "6.9.9",
-  };
-  var options = getMockedGetVersionsOptions({
-    node: version,
-  });
-  check(wanted, options, function(err, result) {
-    t.ifError(err);
-    t.ok(result.versions.node);
-    t.equal(result.versions.node.version.raw, version);
-    t.equal(result.versions.node.wanted.range, "6.9.9");
-    t.equal(result.versions.node.isSatisfied, false);
-    t.equal(result.isSatisfied, false);
+const {
+  nodeCurrent, nodeLTS, nodeOld,
+  npmCurrent, npmLTS, npmLatest, npmOld,
+  npxCurrent, npxLTS, npxLatest, npxOld,
+  yarnCurrent,
+} = require("./test-versions");
+
+const {
+  current, latest, lts,
+  old, npx, yarn,
+} = require("./test-setups");
+
+const {
+  crossTest,
+
+  after,
+  from,
+} = require("./test-helpers");
+
+
+test("global versions only", t => {
+  // check for locally installed npm
+  check({ npm: "3.10.10" }, (error, result) => {
```

Comment:
Would this be reasonable?

```suggestion
  check({ npm: require('./package.json').devDependencies.npm }, (error, result) => {
```

Reply:
Yes, I had that check as well. From the commit message when I changed this:

> It may be a good idea to hard-code this anyway, seeing as the local version might be changed accidentally to coincide with the global one.

I'm not quite sure either way 🤔